packages/core/src/render3/instructions.ts

```diff
@@ -734,19 +732,22 @@ export function directiveCreate<T>(
     }
     previousOrParentNode !.flags = flags;
 
-    ngDevMode && assertDirectivesInRange(index2 - 1);
+    ngDevMode && assertDataInRange(index - 1);
     Object.defineProperty(
         directive, NG_HOST_SYMBOL, {enumerable: false, value: previousOrParentNode});
-    directives[index2] = instance = directive;
-    directives[index2 | 1] = directiveDef;
+    data[index] = instance = directive;
+    ngStaticData[index] = directiveDef !;
```

Comment:
this does not seem right. Should we not be guarding this? We want to avoid the write since we know that we already have the `ngStaticData` initialized. I know we are writing the same data, but it still is memory access, which is slower than reads and which may causes cache misses.

Perhaps we may need a flag such as `staticCreateMode` which would be set whenever the `ngStaticData` is created???